.github/workflows/deploy.yml

```
@@ -13,39 +13,55 @@ on:
 permissions: {}
 
 jobs:
-
-  deploy:
-    if: github.repository == 'pytest-dev/pytest'
-
+  build:
     runs-on: ubuntu-latest
-    timeout-minutes: 30
-    permissions:
-      contents: write
-
+    timeout-minutes: 10
+    environment: deploy
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
         persist-credentials: false
-
     - name: Build and Check Package
       uses: hynek/build-and-inspect-python-package@v1.5
 
+  deploy:
+    if: github.repository == 'pytest-dev/pytest'
+    needs: [build]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    environment: deploy
+    permissions:
+      id-token: write # for pypi oidc publish
+    steps:
     - name: Download Package
       uses: actions/download-artifact@v3
       with:
         name: Packages
         path: dist
-
     - name: Publish package to PyPI
       uses: pypa/gh-action-pypi-publish@release/v1
```

Might as well update to the latest version:

```suggestion
      uses: pypa/gh-action-pypi-publish@v1.8.5
```

(I just tested this version with `pytest-mock` and it worked flawlessly to publish it to test-pypi with trusted-publishers).